test/Makefile

```diff
@@ -3,6 +3,14 @@ ifdef OPENMP
   CFLAGS += -fopenmp
 endif
 
+TARGET := $(shell $(CC) -dumpmachine)
```

Comment:
`uname -s` is not available on windows & tells about the host system. If you're cross-compiling on a linux box & targeting mac, it won't work.
I guess the `cmake` PR makes more sense for true multiplatform builds. Here, I just take care of building properly on MacOS. If for some reason, the shell command fails, then it will fallback to what used to be. Only if it succeeds & you're targeting macOS/iOS will it have any effects.